src/main/kotlin/org/jetbrains/kotlinx/jupyter/repl/impl/InternalEvaluatorImpl.kt

```diff
@@ -57,6 +66,17 @@ internal class InternalEvaluatorImpl(
 
     override val lastClassLoader get() = compiler.lastClassLoader
 
+    override val variablesMap = mutableMapOf<String, VariableState>()
```

Comment:
Difference between variablesCache and variablesState is not clear

Reply:
Perhaps, I should add more comments? The idea was the following: 
-variablesCache stores info about vals 
-variablesMap about everything 
so first is a part of the second, and there is an invariant for them